File: Codes/2610-convert-an-array-into-a-2d-array-with-conditions.java
/*
 * @Author: LetMeFly
 * @Date: 2025-03-19 19:58:16
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-03-19 20:03:10
 */
import java.util.HashMap;
import java.util.Map;
import java.util.ArrayList;
import java.util.List;


class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        List<List<Integer>> ans = new ArrayList<>();
        Map<Integer, Integer> ma = new HashMap<>();
        for (int t : nums) {
            ma.merge(t, 1, Integer::sum);
            if (ma.get(t) > ans.size()) {
                ans.add(new ArrayList<>());
            }
            ans.get(ma.get(t) - 1).add(t);
        }
        return ans;
    }
}

File: Codes/2610-convert-an-array-into-a-2d-array-with-conditions.py
'''
Author: LetMeFly
Date: 2025-03-19 19:56:03
LastEditors: LetMeFly.xyz
LastEditTime: 2025-03-19 19:57:12
'''
from typing import List
from collections import defaultdict


class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        ans: List[List[int]] = []
        ma = defaultdict(int)
        for t in nums:
            ma[t] += 1
            if ma[t] > len(ans):
                ans.append([])
            ans[ma[t] - 1].append(t)
        return ans
